apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/R4ExplanationOfBenefitResourceProviderIT.java

```diff
@@ -900,126 +901,229 @@ public void searchForEobsByMissingPatient() {
   }
 
   /**
-   * Verifies that {@link
-   * gov.cms.bfd.server.war.r4.providers.ExplanationOfBenefitResourceProvider#findByPatient(ca.uhn.fhir.rest.param.ReferenceParam)}
-   * with <code>excludeSAMHSA=true</code> properly filters out SAMHSA-related claims.
+   * Load the SAMPLE_A resources and then tweak each of the claim types that support SAMHSA (all
+   * except PDE) to have a SAMHSA diagnosis code.
    *
-   * @throws FHIRException (indicates test failure)
+   * @return the beneficary record loaded by Sample A
    */
-  @Test
-  public void searchForEobsWithSamhsaFiltering() throws FHIRException {
+  private Beneficiary loadSampleAWithSamhsa() {
     // Load the SAMPLE_A resources normally.
     List<Object> loadedRecords =
         ServerTestUtils.get()
             .loadData(Arrays.asList(StaticRifResourceGroup.SAMPLE_A.getResources()));
 
-    // Tweak the SAMPLE_A Carrier claim such that it's SAMHSA-related.
+    EntityManager entityManager = null;
+
+    try {
+      EntityManagerFactory entityManagerFactory =
+          PipelineTestUtils.get().getPipelineApplicationState().getEntityManagerFactory();
+      entityManager = entityManagerFactory.createEntityManager();
+
+      // Tweak the SAMPLE_A claims such that they are SAMHSA-related.
+      adjustCarrierClaimForSamhsaDiagnosis(loadedRecords, entityManager);
+      adjustInpatientRecordForSamhsaDiagnosis(loadedRecords, entityManager);
+      adjustOutpatientRecordForSamhsaDiagnosis(loadedRecords, entityManager);
+      adjustHhaRecordForSamhsaDiagnosis(loadedRecords, entityManager);
+      adjustSnfRecordForSamhsaDiagnosis(loadedRecords, entityManager);
+      adjustHospiceRecordForSamhsaDiagnosis(loadedRecords, entityManager);
+      adjustDmeRecordForSamhsaDiagnosis(loadedRecords, entityManager);
```

Comment:
Are these only setting up tests for one possible samhsa code?  Would we want to check each place a samhsa code could be present? (ProcedureComp, DiagnosisComp, Line item, both for recognized SAMHSA codes and unknown systems for each)

Reply:
Yes these set up the ITs for Diagnosis checks only. The unit tests more thoroughly go over each of these SAMHSA trigger areas for each of the claim types where they apply.